channeldb/payments.go

```diff
@@ -956,7 +959,14 @@ func (d *DB) DeletePayments(failedOnly, failedHtlcsOnly bool) error {
 		}
 
 		return nil
-	}, func() {})
+	}, func() {
+		numPayments = 0
```

Comment:
this can be `len(deleteIndexes)` instead?

Reply:
No, the `deleteIndexes` variable is only available in the transaction callback and not outside. And I don't really want to change too much in the logic here so a simple counter seemed most straightforward.